### ISSUE TITLE
macOS: implement QuickLook

### ISSUE THREAD

**Issue by formatc1702** [mucommander/mucommander#486]
In macOS Finder, hitting the spacebar allows to "[QuickLook](https://support.apple.com/guide/mac-help/view-and-edit-files-with-quick-look-mh14119/mac)" into the selected file[s].

I am aware that in muCommander, `F3` can be used to view the files.
However:

- the functionality is limited (e.g. cannot easily pinch-zoom images, can't view TIFF files, can't directly annotate images).
- it has some odd bugs (e.g. multi-page PDFs in "single page view continuous" makes the toolbars scroll out of view as you scroll down in the document).
- it feels very slow (compared to QuickLook). 
- I have multiple third-party QuickLook extensions (e.g. Markdown renderer) that I would like to keep using from muCommander.

I would be extremely thankful for an option to choose QuickLook over the built-in viewer. I've seen other third-party software implementing QuickLook, so [there must be some API](https://developer.apple.com/design/human-interface-guidelines/macos/system-capabilities/quick-look/) that can be called.

Thanks for the awesome work!

**Comment by Andreas0602** [comment]
Hi Daniel, you can do it yourself by creating a file ~/Library/Preferences/muCommander/commands.xml with the following content:

```
<?xml version="1.0" encoding="UTF-8"?>
<commands>
    <command alias="Mail"        value="open -a Mail $f"/>
    <command alias="open"        type="system" value="open $f"/>
    <command alias="openURL"     type="system" value="open $f"/>
    <command alias="openFM"      type="system" value="open -a Finder $f" display="Finder"/>
    <command alias="view"        value="qlmanage -p $f" type="invisible"/>
</commands>
```

The line containing “qlmanage” is the relevant line.

**Comment by formatc1702** [comment]
Fantastic! Exactly what I was looking for. I can live with the "DEBUG" in the title ;) Thanks!

**Comment by ahadas** [comment]
> Hi @formatc1702, you can do it yourself by creating a file ~/Library/Preferences/muCommander/commands.xml with the following content: [...]

I wonder if it would make sense to provide such customization per operating system by default
One of the things I had in mind is to package the app differently per operating system (in terms of modules) set keyboard shortcuts per operating system, maybe a complementary part is to include different customization files per-operating system as well

**Comment by Andreas0602** [comment]
IMHO, the default behavior of muCommander when pressing F3 should not be changed. I prefer muCommander's built-in viewer over QuickLook for the following reasons: a) With QuickLook, you cannot search for strings in text files. b) With QuickLook, you cannot select a part of the text and copy it to the clipboard. Both is possible with muCommander's built-in text viewer. For these (and other) reasons, I said in #481 that it would be a pity if the active development of muCommander's built-in viewer/editor would not continue. Just my 2 cents.

**Comment by Andreas0602** [comment]
By the way, the keyboard shortcuts for Find and Save in the built-in text editor/viewer are not configurable at all (neither via the GUI, nor via action_keymap.xml), are they?

**Comment by ahadas** [comment]
> By the way, the keyboard shortcuts for Find and Save in the built-in text editor/viewer are not configurable at all (neither via the GUI, nor via action_keymap.xml), are they?



> By the way, the keyboard shortcuts for Find and Save in the built-in text editor/viewer are not configurable at all (neither via the GUI, nor via action_keymap.xml), are they?

No, [they're hard-coded](https://github.com/mucommander/mucommander/blob/master/mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextViewer.java#L250)

**Comment by Andreas0602** [comment]
> No, [they're hard-coded](https://github.com/mucommander/mucommander/blob/master/mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextViewer.java#L250)

Hmm ... that's bad. It would be good if you could make these shortcuts configurable.

**Comment by meny** [comment]
I would very much appreciate the possibility to use system's <SPACE> key binding to invoke QuickLook while keeping the F3 intact. Is this possible?
